### ISSUE TITLE
bug(rpc): getNonce unexisting contract not detected

### ISSUE THREAD

**Issue by antiyro** [keep-starknet-strange/madara#1373]
Currently if I get a nonce on a non existing contract it returns `0x0`

to reproduce:

```
curl --request POST \
     --url http://localhost:9944 \
     --header 'accept: application/json' \
     --header 'content-type: application/json' \
     --data '{
       "id": 1,
       "jsonrpc": "2.0",
       "method": "starknet_getNonce",
       "params": [
         {"block_number": 2},
         "0x0"
       ]
     }'
```

result

```
{"jsonrpc":"2.0","result":"0x0","id":1}
```

**Comment by Tbelleng** [comment]
Can i take this one sir ?

**Comment by Tbelleng** [comment]
the problem seems to come from this part : https://github.com/keep-starknet-strange/madara/blob/8c285b45101dd20736233408f37d91ab87a995cd/crates/client/storage/src/overrides/schema_v1_override.rs#L126C1-L131C2

In this part, the Nonce is set-up to 0 (so a valid Nonce) even if its do not exist.

**Comment by Tbelleng** [comment]
Fixed by #1374 ✅
